app/models/vehicle_details.rb

```diff
@@ -58,7 +58,7 @@ def taxi_or_phv?
   # Checks if vehicle not taxi or phv and not in list of allowed vehicle types
   # Returns boolean
   def not_taxi_and_correct_type?
-    return false if %w[M3 N1 N2 N3].include?(type_approval&.upcase)
+    %w[M1 M2].include?(type_approval&.upcase) ? (return false) : nil
```

Comment:
I think that below line should also work and is a bit more easy to read and understand:
```suggestion
    return false if %w[M1 M2].include?(type_approval&.upcase)
```

Reply:
We can't, logic was changed, we should show radio button only for `M1` and `M2` types if they not a taxi. Anyway I changed to rails method `exclude` and changed default type to not be m1 or m2.